Signal/src/call/SignalCall.swift

```diff
@@ -39,51 +42,97 @@ protocol CallObserver: class {
 
     // Distinguishes between calls locally, e.g. in CallKit
     let localId: UUID
-    
+
     var hasVideo = false {
         didSet {
-            Logger.debug("\(TAG) hasVideo changed: \(oldValue) -> \(hasVideo)")
-            for observer in observers {
-                observer.value?.hasVideoDidChange(call: self, hasVideo: hasVideo)
+            // This should only occur on the signaling queue.
+            objc_sync_enter(self)
```

Comment:
Ah, as discussed offline, this has to do with synchronizing with the list of observers, which is mutated on the main thread.